grammars/css.cson

```diff
@@ -20,294 +20,1226 @@
+          |ButtonText|CaptionText|GrayText|Highlight|HighlightText|InactiveBorder|InactiveCaption
+          |InactiveCaptionText|InfoBackground|InfoText|Menu|MenuText|Scrollbar|ThreeDDarkShadow
+          |ThreeDFace|ThreeDHighlight|ThreeDLightShadow|ThreeDShadow|Window|WindowFrame|WindowText)
+          (?![\\w-])
+        '''
+        'name': 'invalid.deprecated.color.system.css'
+      }
+    ]
+  'combinators':
+    'patterns': [
+      {
+        'match': '/deep/|>>>'
+        'name': 'invalid.deprecated.combinator.css'
+      }
+      {
+        'match': '>>|>|\\+|~'
+        'name': 'keyword.operator.combinator.css'
+      }
+    ]
+  'commas':
+    'match': ','
+    'name': 'punctuation.separator.list.comma.css'
+  'comment-block':
+    'begin': '/\\*'
+    'beginCaptures':
       '0':
-        'name': 'keyword.control.at-rule.charset.css'
-      '1':
-        'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=;|$))'
-    'name': 'meta.at-rule.charset.css'
+        'name': 'punctuation.definition.comment.begin.css'
+    'end': '\\*/'
+    'endCaptures':
+      '0':
+        'name': 'punctuation.definition.comment.end.css'
+    'name': 'comment.block.css'
+  'escapes':
     'patterns': [
       {
-        'include': '#string-double'
+        'match': '\\\\[0-9a-fA-F]{1,6}'
+        'name': 'constant.character.escape.codepoint.css'
       }
       {
-        'include': '#string-single'
+        'match': '\\\\$\\n?'
+        'name': 'constant.character.escape.newline.css'
       }
       {
-        'include': '#comment-block'
+        'match': '\\\\.'
+        'name': 'constant.character.escape.css'
       }
     ]
-  }
-  {
-    'begin': '(@)import\\b'
-    'captures':
+  'feature-query':
+    'begin': '\\('
+    'beginCaptures':
+      '0':
+        'name': 'punctuation.definition.condition.begin.bracket.round.css'
+    'end': '\\)'
+    'endCaptures':
       '0':
-        'name': 'keyword.control.at-rule.import.css'
+        'name': 'punctuation.definition.condition.end.bracket.round.css'
+    'name': 'meta.feature-query.css'
+    'patterns': [
+      {
+        'include': '#feature-query-operators'
+      }
+      {
+        'include': '#feature-query'
+      }
+    ]
+  'feature-query-operators':
+    'patterns': [
+      {
+        'match': '(?i)(?<=[\\s()]|^|\\*/)(and|not|or)(?=[\\s()]|$|/\\*)'
+        'name': 'keyword.operator.logical.feature.$1.css'
+      }
+      {
+        'include': '#rule-list-innards'
+      }
+    ]
+  'font-features':
+    'begin': '''(?xi)
+      ((@)(annotation|character-variant|ornaments|styleset|stylistic|swash))
+      (?=[\\s{@'";]|/\\*|$)\\s*
+    '''
+    'beginCaptures':
       '1':
+        'name': 'keyword.control.at-rule.${3:/downcase}.css'
+      '2':
         'name': 'punctuation.definition.keyword.css'
-    'end': '\\s*((?=;|\\}))'
-    'name': 'meta.at-rule.import.css'
+    'end': '(?<=})\\s*'
```

Comment:
No, it's not surprising at all. It's a simple precedence issue. When selecting the pattern to match an expression, TextMate will use whatever matches earliest, which means the `\s*` forces priority over other patterns that might terminate at a closing `}`.

Can you stop introducing new problems, please? Whitespace isn't going to kill anybody, that's has dragged on for ages, and hell, if you want perfect grammars, I'd suggest focussing on the others that **actually need them**.